pysteps/blending/steps.py

```diff
@@ -2043,78 +2089,25 @@ def _compute_cascade_decomposition_radar(
-        precip_models_cascade = []
-        # Loop through the n_models
-        for i in range(precip_models.shape[0]):
-            precip_model_cascade = []
-            # Loop through the time steps
-            for j in range(precip_models.shape[1]):
-                precip_model_cascade.append(
-                    decompositor(
-                        field=precip_models[i, j, :, :],
-                        bp_filter=bp_filter,
-                        fft_method=fft,
-                        output_domain=domain,
-                        normalize=True,
-                        compute_stats=True,
-                        compact_output=True,
-                    )
-                )
-            precip_models_cascade.append(precip_model_cascade)
-        precip_models_cascade = np.array(precip_models_cascade)
+def _compute_cascade_recomposition_nwp(precip_models, recompositor):
+    """If necessary, recompose (NWP) model forecasts."""
+    precip_models_cascade = precip_models
+    precip_models_pm = None
 
-        precip_models, precip_model_cascade = None, None
-    else:
-        precip_models_cascade = precip_models
-        precip_models_pm = None
-
-    # Stack the (NWP) model cascades in separate normalized cascades and return
-    # the means and sigmas.
-    # The normalized model cascade should have the shape:
-    # [n_models, n_timesteps, n_cascade_levels, m, n]
-    precip_models = []
-    mu_models = []
-    sigma_models = []
-    # Stack it per model and combine that
+    # Recompose the (NWP) model cascades to have rainfall fields per
+    # model and time step, which will be used in the probability matching steps.
+    # Recomposed cascade will have shape: [n_models, n_timesteps, m, n]
+    precip_models_pm = []
     for i in range(precip_models_cascade.shape[0]):
-        precip_model, mu_model, sigma_model = blending.utils.stack_cascades(
-            R_d=precip_models_cascade[i, :], donorm=False
-        )
-        precip_models.append(precip_model)
-        mu_models.append(mu_model)
-        sigma_models.append(sigma_model)
-
-    precip_models = np.stack(precip_models)
-    mu_models = np.stack(mu_models)
-    sigma_models = np.stack(sigma_models)
-
-    precip_model, mu_model, sigma_model = None, None, None
-
-    if precip_models_pm is None:
-        # Finally, recompose the (NWP) model cascades to have rainfall fields per
-        # model and time step, which will be used in the probability matching steps.
-        # Recomposed cascade will have shape: [n_models, n_timesteps, m, n]
-        precip_models_pm = []
-        for i in range(precip_models_cascade.shape[0]):
-            precip_model_pm = []
-            for time_step in range(precip_models_cascade.shape[1]):
-                precip_model_pm.append(
-                    recompositor(precip_models_cascade[i, time_step])
-                )
-            precip_models_pm.append(precip_model_pm)
+        precip_model_pm = []
```

Comment:
Is precip_model_pm still the most logical naming here, since you have change it in the main loop to ``precip_models``?

Reply:
Yes I thought of this as well, I guess I can update it everywhere in the file where there is still precip_models_pm